### PR TITLE
feat(fields): ClickUp fields dependency tracking [CLK-500346]

### PR DESCRIPTION
**Pull request by mysza**
## Provides means to determine dependent formulas

When a formula definition changes, we need to:

1. validate the formula, i.e. test if it has no circular references (formula A depending on B, B depending on C, C depending on A) and if the nesting level is not higher than allowed
2. figure out what other formulas might depend on the changed one, if we have the formula-in-formula functionality
 
For this, we have additional class `ClickUpFieldsDependencyTracker` allowing for finding the formulas that depend on the changed formula. With this knowledge, the client can recalculate the values of the fields affected.

The dependency detector builds a directed graph of the custom fields variables, and uses DFS to find the variables dependent on the requested one. It is also capable of determining if there are cycles in the graph, and guard against too high nesting level.

# Tests
- all current tests pass
- new functionality is covered with unit tests
- added two functions testing the performance of the dependencies and cycles detection based on pre-generated data (a list of variables). The data generation is also implemented as a script
- the performance seems to be good enough to not need to use database for storing dependencies: with 30k custom fields variables (50% more than our biggest workspace), with 50% of which are formulas, each formula depending on 10 other fields, of which 50% are other formulas (quite extreme example), the time to validate the graph is around 50ms on my machine (graph construction and dependents lookup is much faster, validation is the only one traversing the whole graph).
- also, if we ever decide to store the dependencies, backfilling should be rather simple (on read, if not yet checked, do the dependency check and store the value)

### CODE REVIEW

File: src/clickup/clickupFieldsDependencyTracker.ts [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+}
+
+interface DependencyGraphNode {
+    dependents: FieldName[];
+    dependencies: FieldName[];
+}
+
+interface ValidationContext {
+    maxLevels: number;
+    recStack: Set<FieldName>;
```
Comment (nikitaclicks) [inline src/clickup/clickupFieldsDependencyTracker.ts:19]: what's `rec`?

Comment (mysza) [reply src/clickup/clickupFieldsDependencyTracker.ts:19]: recursive/recursion, keeps field names within the same path for circular dependency tracking.

File: test/data/test_custom_fields.json
Comment (szygendab) [inline test/data/test_custom_fields.json:1]: I wonder if we need to keep such a large file in git? Cannot it be generated on-demand with zgeneratePerfTestData.ts`?

**Review by szygendab — APPROVED** [review]

File: test/data/test_custom_fields.json
Comment (mysza) [reply test/data/test_custom_fields.json:1]: It can, I'll remove it from the git.

**Pull request merged**